src/commands/build.ts

```diff
@@ -124,207 +88,120 @@ export async function command(commandOptions: CommandOptions) {
+    const allFiles = glob.sync(`**/*`, {
+      ignore: config.exclude,
+      cwd: dirDisk,
+      absolute: true,
+      nodir: true,
+      dot: true,
+    });
+    const allBuildNeededFiles: string[] = [];
+    await Promise.all(
+      allFiles.map(async (f) => {
+        f = path.resolve(f); // this is necessary since glob.sync() returns paths with / on windows.  path.resolve() will switch them to the native path separator.
+        const {baseExt} = getExt(f);
+        allBuildNeededFiles.push(f);
+      }),
+    );
+    includeFileSets.push([dirDisk, dirDest, allBuildNeededFiles]);
   }
 
   const allBuiltFromFiles = new Set<string>();
   const allFilesToResolveImports: SnowpackBuildMap = {};
-  for (const workerConfig of relevantWorkers) {
-    const {id, match, type} = workerConfig;
-    if (type !== 'build' || match.length === 0) {
-      continue;
-    }
 
-    messageBus.emit('WORKER_UPDATE', {id, state: ['RUNNING', 'yellow']});
-    for (const [dirDisk, dirDest, allFiles] of includeFileSets) {
-      for (const locOnDisk of allFiles) {
-        const inputExt = getExt(locOnDisk);
-        if (!match.includes(inputExt.baseExt) && !match.includes(inputExt.expandedExt)) {
-          continue;
-        }
-        const fileContents = await fs.readFile(locOnDisk, {encoding: 'utf8'});
-        let fileBuilder = getFileBuilderForWorker(cwd, workerConfig, messageBus);
-        if (!fileBuilder) {
-          continue;
-        }
-        let outLoc = locOnDisk.replace(dirDisk, dirDest);
-        const extToReplace = srcFileExtensionMapping[inputExt.baseExt];
-        if (extToReplace) {
-          outLoc = outLoc.replace(new RegExp(`\\${inputExt.baseExt}$`), extToReplace!);
-        }
-
-        const builtFile = await fileBuilder({
-          contents: fileContents,
-          filePath: locOnDisk,
-          isDev: false,
-        });
-        if (!builtFile) {
-          continue;
-        }
-        let {result: code, resources} = builtFile;
-        const urlPath = outLoc.substr(dirDest.length + 1);
-        for (const plugin of config.plugins) {
-          if (plugin.transform) {
-            code =
-              (await plugin.transform({contents: fileContents, urlPath, isDev: false}))?.result ||
-              code;
+  for (const [dirDisk, dirDest, allFiles] of includeFileSets) {
+    for (const locOnDisk of allFiles) {
+      const {baseExt: fileExt} = getExt(locOnDisk);
+      let outLoc = locOnDisk.replace(dirDisk, dirDest);
+      let builtLocOnDisk = locOnDisk;
+      const extToReplace = srcFileExtensionMapping[fileExt];
+      if (extToReplace) {
+        outLoc = replaceExt(outLoc, extToReplace);
+        builtLocOnDisk = replaceExt(builtLocOnDisk, extToReplace);
+      }
+      const fileContents = await fs.readFile(locOnDisk, getEncodingType(extToReplace || fileExt));
+      let builtFileOutput: SnowpackBuildMap;
+      builtFileOutput = await buildFile(locOnDisk, fileContents, {
```

Comment:
issue for a follow up PR: buildFile() is using `SnowpackBuildMap` as its return type. This data structure has some similarities with what we're trying to do, but it is really meant to represent a file on disk. It uses the full file path as the key, has lots of info about "locOnDisk" and file extensions, etc.  

You can see us doing a lot of duplicate logic above and below to map the source file to both the final build location and the "fake" built files with the full path to the source directory.

Instead, this should use a new data type that purely represents the file build, something like:

```
builtFileOutput = {
  ".js": {contents: '...'},
  ".css": {contents: '...'},
}
```

This would simplify lots of smaller bits of logic and remove a couple of footguns related to using a "file on disk" data representation to represent fake files.

Reply:
Good call out. Especially on the “footguns” 😄 

I think we’ll have flexibility to change this now that we’re just using one predictable TS interface—hopefully as we change it, TS will do most of that work for us.